src/web/client/dal/fileSystemProvider.ts

```diff
@@ -310,6 +310,99 @@ export class PortalsFS implements vscode.FileSystemProvider {
         return files;
     }
 
+    async searchTextResults(query: vscode.TextSearchQuery, options: vscode.TextSearchOptions, progress: vscode.Progress<vscode.TextSearchResult>) {
+        const results = await this.searchText(
+            query,
+            { maxResults: options.maxResults }
+        );
+
+        if (results === undefined) {
+            return { limitHit: true };
+        }
+
+        for (const match of results.matches) {
+            progress.report({
+                uri: match.uri,
+                ranges: match.ranges,
+                preview: {
+                    text: match.preview,
+                    matches: match.matches,
+                },
+            });
+        }
+
+        return { limitHit: false };
+    }
+
+    async searchText(query: vscode.TextSearchQuery, options: { maxResults?: number }): Promise<ISearchQueryResults> {
+        const matches: ISearchQueryMatch[] = [];
+        const files = await this.iterateDirectory(WebExtensionContext.rootDirectory);
+
+        // counter to keep track of the number of matches
+        let counter = 0;
+        let match: ISearchQueryMatch;
+        for (const file of files) {
+            const content = await this.readFile(file);
+
+            // Convert buffer to string and replace windows line endings with unix line endings
+            const text = new TextDecoder().decode(content).replace(/\r\n/g, '\n');
+
+            // Convert windows line endings with unix line endings
+            const pattern = query.pattern.replace(/\r\n/g, '\n');
+
+            const lines = text.split('\n');
+
+            match = {
+                uri: file,
+                ranges: [],
+                preview: text,
+                matches: [],
+            };
+
+            let regex;
+            if (query.isWordMatch) {
+                // \b is a word boundary
+                regex = query.isCaseSensitive ? new RegExp(`\\b${pattern}\\b`) : new RegExp(`\\b${pattern}\\b`, "i");
```

Comment:
we should add telemetry to meaure performance here

Reply:
done!

Reply:
PLease make sure what we are logging is compliant and this will not have any performance impact